ItemLevelDownloader/Providers/webProvider.ts

```diff
@@ -0,0 +1,115 @@
+﻿import * as http from 'http';
+import * as https from 'https';
+import * as url from 'url';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as stream from 'stream';
+
+var handlebars = require('handlebars');
+
+import * as models from '../Models';
+
+export class WebProvider implements models.IArtifactProvider {
+
+    constructor(rootItemsLocation, templateFile: string, username: string, password: string, variables: any) {
+        this._rootItemsLocation = rootItemsLocation;
+        this._templateFile = templateFile;
+        this._username = username;
+        this._password = password;
+        this._variables = variables;
+    }
+
+    getRootItems(): Promise<models.ArtifactItem[]> {
+        return this.getItems(this._rootItemsLocation);
+    }
+
+    getArtifactItems(artifactItem: models.ArtifactItem): Promise<models.ArtifactItem[]> {
+        var itemsUrl = artifactItem.metadata["downloadUrl"];
+        return this.getItems(itemsUrl);
```

Comment:
default timeout is 2 minutes, after 2 mins timeout event will be raised by nodejs it's upto us if we want to abort the request. I had added a todo to figure out what to do here.